src/ctapipe/image/modifications.py

```diff
@@ -100,6 +108,177 @@ def _smear_psf_randomly(
     return new_image
 
 
+class WaveformModifier(TelescopeComponent):
+    """
+    Component to add NSB noise to R1 waveforms (intended in principle to be
+    applied on MC simulations, to make them closer to real data)
+    The noise waveforms are read from a dedicated sim_telarray file, which
```

Comment:
After a brief look over the code, it seems we are not bound to simulations as a noise source.
I would change `from a dedicated sim_telarray file` to `from a dedicated NSB noise file (simulated or pedestal calibration)` and add event type filtering to select pedestal events (we are not 100% sure whether pedestal and flasher will come in the same or in different files).